best-practices/client-side/framework/react/components/media-cropper/src/components/media-cropper/index.js

```diff
@@ -0,0 +1,204 @@
+import React, { Component } from 'react';
+
+import { ImageRenderer, VideoRenderer } from './renderers';
+
+export default class MediaCropper extends Component {
+    get isVideo() {
+        return this.state.isVideo;
+    }
+
+    get croppedMedia() {
+        return this.refs.media.croppedMedia;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            zoom: 1,
+            transX: 0,
+            transY: 0,
+            media: null,
+            isVideo: false,
+            dragging: false,
+            mouseStart: {
+                x: 0,
```

Comment:
Nothing wrong with using Class-based components, but the more modern React projects we are working on currently use the newer [hooks](https://reactjs.org/docs/hooks-intro.html) to manage state within a component.

Reply:
I'll look into converting this, I think my Java background leads me to defaulting to class-based solutions.